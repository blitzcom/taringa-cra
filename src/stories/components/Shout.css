.Shout {
  display: flex;
}

.Shout-thumbnail {
  height: 70px;
  margin-right: 16px;
  width: 70px;
  border: 1px solid #eaeaea;
  border-radius: 3px;
}

.Shout-thumbnail > img {
  border-radius: 3px;
  height: 100%;
  object-fit: cover;
  width: 68px;
}

.Shout-thumbnail:hover {
  text-decoration: none;
}

.Shout-meta {
  color: #b6b6b6;
  font-size: 12px;
  margin-bottom: 0;
}

.Shout-meta > a {
  color: black;
  margin: 0 3px;
}
