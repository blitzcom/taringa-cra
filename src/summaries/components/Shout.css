.Shout {
  width: 100%;
}

.Shout-head {
  display: flex;
  flex-direction: row;
  padding: 8px;
}

.Shout-preview {
  background: #f0f0f0;
  border-radius: 0 0 3px 3px;
  margin: 0 0px;
  padding: 8px;
  text-align: center;
}

.Shout-preview > img {
  max-height: 500px;
  max-width: 80%;
}

.Shout-thumbnail {
  border-radius: 3px;
  border: 1px solid #eaeaea;
  height: 70px;
  margin-right: 16px;
  min-width: 70px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 70px;
}

.Shout-thumbnail-icon {
  color: #f0f0f0;
}

.Shout-thumbnail > img {
  border-radius: 3px;
  height: 100%;
  object-fit: cover;
  width: 68px;
}

.Shout-thumbnail:hover {
  text-decoration: none;
}

.Shout-meta {
  color: #b6b6b6;
  font-size: 12px;
  margin-bottom: 0;
}

.Shout-meta > a {
  color: black;
  margin: 0 3px;
}
