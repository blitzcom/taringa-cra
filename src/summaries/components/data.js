export const SummaryShout = {
  id: "z50xj",
  created: 1513920502005,
  edited: null,
  title: "",
  nsfw: false,
  classic: {
    tid: "ctLMo",
    type: "shout"
  },
  tags: [
    "image",
    "classic"
  ],
  slug: "tienen-que-shoutear-minas-asi_z50xj",
  type: "story:summary",
  summary: {
    link: null,
    images: {
      slice: [
        {
          type: "image",
          url: "https://k60.kn3.net/taringa/1/8/3/5/9/3/15/f3d3_14/61D.jpg",
          width: 640,
          height: 774,
          format: "jpeg"
        }
      ],
      amount: 1
    },
    excerpt: "Tienen que shoutear minas asi",
    textLength: 29
  },
  channel: {
    type: "channel:summary",
    id: "1ld",
    created: "2016-08-05T03:09:30.000Z",
    name: "user-F3D3_14",
    title: "user-F3D3_14",
    channelType: "user_feed",
    thumbnail: "",
    background: "",
    description: "",
    category: "unknown/unknown",
    owner: {
      id: "1lc",
      created: "2016-08-05T03:09:30.000Z",
      type: "user",
      tid: 18359315,
      username: "F3D3_14",
      firstname: "Fedeformito el",
      lastname: "Perro CÃ³smico",
      birthday: "1998-1-8",
      avatar: "https://a06.t26.net/taringa/avatares/0/3/6/8/F/7/F3D3_14/120x120_885.jpg",
      site: "",
      gender: "m",
      country: "AR",
      message: "ðŸ˜œ Altoke Perro, te likeo hasta el corazÃ³n",
      migrated: [
        "classicCategories",
        "followers"
      ],
      subscribedToDefaultChannels: true
    },
    subscribers: 0,
    stories: 1753,
    latestStory: "2018-01-22T20:04:02.000Z",
    state: {
      subscription: "unsubscribed"
    }
  },
  owner: {
    id: "1lc",
    username: "F3D3_14",
    type: "user:summary",
    firstname: "Fedeformito el",
    lastname: "Perro CÃ³smico",
    avatar: "https://a06.t26.net/taringa/avatares/0/3/6/8/F/7/F3D3_14/120x120_885.jpg",
    gender: "m",
    country: "AR"
  },
  upvotes: 13799,
  downvotes: 100,
  shares: 4500,
  comments: 8000,
  state: {
    voted: 0
  },
  brandSafe: false,
  listViewSafe: false
}
