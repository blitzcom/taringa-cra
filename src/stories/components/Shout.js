import React from 'react'
import TimeAgo from 'react-timeago'

import './Shout.css'
import { humanizeNum, esFormatter } from '../../Utils'

const Shout = props => {
  return (
    <div className="Shout">
      <a className="Shout-thumbnail" href="/">
        {
          props.summary.images.amount > 0 && (
            <img
              src={props.summary.images.slice[0].url}
              alt="Thumbnail"
            />
          )
        }
      </a>
      <div className="Shout-info">
        <a
          className="Shout-title text-dark"
          href={`/story/${props.slug}`}
        >
          { props.summary.excerpt }
        </a>
        <p className="Shout-meta">
          Posteado por
          <a href={`/user/${props.owner.username}`}>
            {props.owner.username}
          </a>
          <TimeAgo
            date={props.created}
            formatter={esFormatter}
          />
        </p>
        <div className="Shout-actions">
          <button
            className="btn btn-action"
            title="Expandir"
          >
            <i className="fa fa-expand"/>
          </button>

          <button
            className="btn btn-action"
            title="Comentarios"
          >
            <i className="fa fa-comments"/>
            {humanizeNum(props.comments)} comentarios
          </button>

          <button
            className="btn btn-action"
            title="Compartir"
          >
            <i className="fa fa-retweet"/>
            {humanizeNum(props.shares)}
          </button>
        </div>
      </div>
    </div>
  )
}

export default Shout
