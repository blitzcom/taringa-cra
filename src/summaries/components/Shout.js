import React, { Component } from 'react'
import TimeAgo from 'react-timeago'

import './Shout.css'
import Score from './Score'
import Action from '../../common/Action'
import ToggleAction from '../../common/ToggleAction'
import { humanizeNum, esFormatter } from '../../Utils'

class Shout extends Component {
  constructor (props) {
    super(props)
    this.state = {
      previewIsOpen: props.previewIsOpen || false
    }
  }

  togglePreview () {
    const { previewIsOpen } = this.state

    this.setState({ previewIsOpen: !previewIsOpen })
  }

  render () {
    const { previewIsOpen } = this.state

    return (
      <div className="Shout">
        <div className="Shout-head">
          <Score
            downvotes={this.props.downvotes}
            onVoteDown={this.props.onVoteDown}
            onVoteUp={this.props.onVoteUp}
            upvotes={this.props.upvotes}
            voted={this.props.voted}
            isVoting={this.props.isVoting}
          />

          <a className="Shout-thumbnail" href="/">
            {
              this.props.summary.images.amount > 0 ? (
                <img
                  src={this.props.summary.images.slice[0].url}
                  alt="Thumbnail"
                />
              ) : (
                <i className="fa fa-comment fa-2x Shout-thumbnail-icon"/>
              )
            }
          </a>

          <div className="Shout-info">
            <a
              className="Shout-title text-dark"
              href={`/story/${this.props.slug}`}
            >
              { this.props.summary.excerpt }
            </a>

            <p className="Shout-meta">
              Posteado por
              <a href={`/user/${this.props.owner.username}`}>
                {this.props.owner.username}
              </a>
              <TimeAgo
                date={this.props.created}
                formatter={esFormatter}
              />
            </p>

            <div className="Shout-actions">
              {
                this.props.summary.images.amount > 0 && (
                  <ToggleAction
                    activeIcon="compress"
                    inactiveIcon="expand"
                    isToggled={previewIsOpen}
                    onClick={this.togglePreview.bind(this)}
                    title="Vista previa"
                  />
                )
              }

              <Action icon="comments" title="Comentar">
                { humanizeNum(this.props.comments) } comentarios
              </Action>

              <Action icon="retweet" title="Compartir">
                { humanizeNum(this.props.shares) }
              </Action>
            </div>
          </div>
        </div>
        {
          previewIsOpen && (
            <div className="Shout-preview">
              <img
                src={this.props.summary.images.slice[0].url}
                alt="Preview"
              />
            </div>
          )
        }
      </div>
    )
  }
}

export default Shout
