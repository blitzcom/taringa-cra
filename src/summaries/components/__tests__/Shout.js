import React from 'react'
import renderer from 'react-test-renderer'

import { StoryShout } from '../data'
import Shout from '../Shout'

describe('Shout', () => {
  beforeEach(() => {
    Date.now = jest.fn(() => 1513924592705)
  })

  it('renders default', () => {
    const data = StoryShout
    data.summary.images.amount = 0

    const tree = renderer.create(<Shout {...StoryShout} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with image thumbnail', () => {
    const data = StoryShout
    data.summary.images.amount = 1

    const tree = renderer.create(<Shout {...StoryShout} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with image thumbnail and preview', () => {
    const data = StoryShout
    data.summary.images.amount = 1

    const tree = renderer
      .create(<Shout {...StoryShout} previewIsOpen />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
