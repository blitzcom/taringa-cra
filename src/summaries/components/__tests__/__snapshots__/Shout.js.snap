// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shout renders default 1`] = `
<div
  className="Shout"
>
  <div
    className="Shout-head"
  >
    <div
      className="Score-display"
    >
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-up"
        />
      </button>
      <div
        className="Score-votes"
      >
        13.7k
      </div>
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-down"
        />
      </button>
    </div>
    <a
      className="Shout-thumbnail"
      href="/"
    >
      <i
        className="fa fa-comment fa-2x Shout-thumbnail-icon"
      />
    </a>
    <div
      className="Shout-info"
    >
      <a
        className="Shout-title text-dark"
        href="/story/tienen-que-shoutear-minas-asi_z50xj"
      >
        Tienen que shoutear minas asi
      </a>
      <p
        className="Shout-meta"
      >
        Posteado por
        <a
          href="/user/F3D3_14"
        >
          F3D3_14
        </a>
        <time
          dateTime="2017-12-22T05:28:22.005Z"
          title="2017-12-22 05:28"
        >
          hace 1 hora
        </time>
      </p>
      <div
        className="Shout-actions"
      >
        <button
          className="btn btn-action"
          title="Comentar"
        >
          <i
            className="fa fa-comments"
          />
          8k
           comentarios
        </button>
        <button
          className="btn btn-action"
          title="Compartir"
        >
          <i
            className="fa fa-retweet"
          />
          4.5k
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Shout renders with image thumbnail 1`] = `
<div
  className="Shout"
>
  <div
    className="Shout-head"
  >
    <div
      className="Score-display"
    >
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-up"
        />
      </button>
      <div
        className="Score-votes"
      >
        13.7k
      </div>
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-down"
        />
      </button>
    </div>
    <a
      className="Shout-thumbnail"
      href="/"
    >
      <img
        alt="Thumbnail"
        src="https://k60.kn3.net/taringa/1/8/3/5/9/3/15/f3d3_14/61D.jpg"
      />
    </a>
    <div
      className="Shout-info"
    >
      <a
        className="Shout-title text-dark"
        href="/story/tienen-que-shoutear-minas-asi_z50xj"
      >
        Tienen que shoutear minas asi
      </a>
      <p
        className="Shout-meta"
      >
        Posteado por
        <a
          href="/user/F3D3_14"
        >
          F3D3_14
        </a>
        <time
          dateTime="2017-12-22T05:28:22.005Z"
          title="2017-12-22 05:28"
        >
          hace 1 hora
        </time>
      </p>
      <div
        className="Shout-actions"
      >
        <button
          className="btn btn-action"
          onClick={[Function]}
          title="Vista previa"
        >
          <i
            className="fa fa-expand"
          />
        </button>
        <button
          className="btn btn-action"
          title="Comentar"
        >
          <i
            className="fa fa-comments"
          />
          8k
           comentarios
        </button>
        <button
          className="btn btn-action"
          title="Compartir"
        >
          <i
            className="fa fa-retweet"
          />
          4.5k
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Shout renders with image thumbnail and preview 1`] = `
<div
  className="Shout"
>
  <div
    className="Shout-head"
  >
    <div
      className="Score-display"
    >
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-up"
        />
      </button>
      <div
        className="Score-votes"
      >
        13.7k
      </div>
      <button
        className="Score-vote"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-down"
        />
      </button>
    </div>
    <a
      className="Shout-thumbnail"
      href="/"
    >
      <img
        alt="Thumbnail"
        src="https://k60.kn3.net/taringa/1/8/3/5/9/3/15/f3d3_14/61D.jpg"
      />
    </a>
    <div
      className="Shout-info"
    >
      <a
        className="Shout-title text-dark"
        href="/story/tienen-que-shoutear-minas-asi_z50xj"
      >
        Tienen que shoutear minas asi
      </a>
      <p
        className="Shout-meta"
      >
        Posteado por
        <a
          href="/user/F3D3_14"
        >
          F3D3_14
        </a>
        <time
          dateTime="2017-12-22T05:28:22.005Z"
          title="2017-12-22 05:28"
        >
          hace 1 hora
        </time>
      </p>
      <div
        className="Shout-actions"
      >
        <button
          className="btn btn-action active"
          onClick={[Function]}
          title="Vista previa"
        >
          <i
            className="fa fa-compress"
          />
        </button>
        <button
          className="btn btn-action"
          title="Comentar"
        >
          <i
            className="fa fa-comments"
          />
          8k
           comentarios
        </button>
        <button
          className="btn btn-action"
          title="Compartir"
        >
          <i
            className="fa fa-retweet"
          />
          4.5k
        </button>
      </div>
    </div>
  </div>
  <div
    className="Shout-preview"
  >
    <img
      alt="Preview"
      src="https://k60.kn3.net/taringa/1/8/3/5/9/3/15/f3d3_14/61D.jpg"
    />
  </div>
</div>
`;
